src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -797,9 +801,17 @@ public static CQLStatement.Raw parseStatement(String queryStr) throws SyntaxExce
         }
     }
 
-    private static int measure(Object key, Prepared value)
+    private static int measurePstmnt(Prepared value)
     {
-        return Ints.checkedCast(ObjectSizes.measureDeep(key) + ObjectSizes.measureDeep(value));
+        return Ints.checkedCast(ObjectSizes.measureDeep(value));
+    }
+
+    private static int getSizeForCache(MD5Digest key, Prepared value)
+    {
+        if (value.pstmntSize < 0)
```

Comment:
is this actually ever going to happen? I think `assert value.pstmntSize > 0` is enough. `> 0` because I don't think having size of 0 makes sense. It is invalid size too.

Reply:
Yes this is possible for yet non-parsed/prepared statements, DTO pstmnts, etc So I think I'll leave it as it is.